instrumentation/servlet/servlet-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/v3_0/snippet/SnippetInjectingResponseWrapper.java

```diff
@@ -52,7 +60,7 @@ public boolean containsHeader(String name) {
   public void setHeader(String name, String value) {
     if (isContentTypeTextHtml() && "Content-Length".equalsIgnoreCase(name)) {
```

Comment:
```suggestion
    // checking content-type is just an optimization to avoid unnecessary parsing
    if ("Content-Length".equalsIgnoreCase(name) && isContentTypeTextHtml()) {
```